src/Raven.Server/Documents/Indexes/Persistence/Lucene/LuceneIndexWriter.cs

```diff
@@ -36,82 +29,41 @@ public class LuceneIndexWriter : IDisposable
 
         private readonly IndexWriter.MaxFieldLength maxFieldLength;
 
-        private int currentNumberOfWrites;
-
         private readonly IndexWriter.IndexReaderWarmer _indexReaderWarmer;
 
-        private int changesSinceCommit;
-
-        private readonly Field forceCommitField = new Field("__dummy_force_index_write", "forced", Field.Store.NO, Field.Index.NOT_ANALYZED_NO_NORMS);
-        private readonly Term forceCommitTerm = new Term("__dummy_force_index_write", "forced");
-        private readonly global::Lucene.Net.Documents.Document forceCommitDoc;
-        private DateTime lastCommitDataStoreTime;
-
-        public Directory Directory
-        {
-            get
-            {
-                if (indexWriter != null)
-                    return indexWriter.Directory;
-
-                return null;
-            }
-        }
-
-        public Analyzer Analyzer
-        {
-            get
-            {
-                if (indexWriter != null)
-                    return indexWriter.Analyzer;
+        public Directory Directory => indexWriter?.Directory;
 
-                return null;
-            }
-        }
+        public Analyzer Analyzer => indexWriter?.Analyzer;
 
-        public LuceneIndexWriter(Directory d, Analyzer a, IndexDeletionPolicy deletionPolicy, IndexWriter.MaxFieldLength mfl, int maximumNumberOfWritesBeforeRecreate, IndexWriter.IndexReaderWarmer indexReaderWarmer)
+        public LuceneIndexWriter(Directory d, Analyzer a, IndexDeletionPolicy deletionPolicy, IndexWriter.MaxFieldLength mfl, IndexWriter.IndexReaderWarmer indexReaderWarmer)
         {
             directory = d;
             analyzer = a;
             indexDeletionPolicy = deletionPolicy;
             maxFieldLength = mfl;
             _indexReaderWarmer = indexReaderWarmer;
-            this.maximumNumberOfWritesBeforeRecreate = maximumNumberOfWritesBeforeRecreate;
-
-            forceCommitDoc = new global::Lucene.Net.Documents.Document();
-            forceCommitDoc.Add(forceCommitField);
 
-            RecreateIfNecessary(force: true);
+            RecreateIndexWriter();
```

Comment:
Paweł's profiling didn't showed it to be very expensive. Also it appears we don't have any other option since Lucene keeps things in memory, but VoronIndexInput is accessible only when it has a relevant read transaction. We were getting AccessViolationException if we didn't force the index writer recreation every time we commit anything- I already talked to you about it.